src/modules/python/pbs_hooks/PBS_cray_jacs.PY

```diff
@@ -0,0 +1,339 @@
+    r = requests.delete(url, **kwargs)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received status code = %s' % r.status_code)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received text %s' % r.text)
+    return r
+
+
+def caller_name(frames=1):
+    """
+    Return the name of the nth calling function or method.
+    """
+    return str(sys._getframe(frames).f_code.co_name)
+
+
+def log_function_name():
+    """
+    Log the caller's name
+    """
+    pbs.logmsg(pbs.EVENT_DEBUG4, '%s: Method called' % caller_name(2))
+
+
+def log_with_caller(sev, mes):
+    """
+    Wrapper to pbs.logmsg with caller's name prepended
+    """
+    pbs.logmsg(sev, '%s: %s' % (caller_name(2), mes))
+
+
+def merge_dict(base, new):
+    """
+    Merge together two multilevel dictionaries where new
+    takes precedence over base
+    """
+    if not isinstance(base, dict):
+        raise ValueError('base must be type dict')
+    if not isinstance(new, dict):
+        raise ValueError('new must be type dict')
+    newkeys = new.keys()
+    merged = {}
+    for key in base:
+        if key in newkeys and isinstance(base[key], dict):
+            # Take it off the list of keys to copy
+            newkeys.remove(key)
+            merged[key] = merge_dict(base[key], new[key])
+        else:
+            merged[key] = copy.deepcopy(base[key])
+    # Copy the remaining unique keys from new
+    for key in newkeys:
+        merged[key] = copy.deepcopy(new[key])
+    return merged
+
+
+def handle_execjob_begin():
+    """
+    Handler for execjob_begin events.
+    """
+    log_function_name()
+    event = HookHelper.get_event()
+    jid = event.job.id
+    uid = pwd.getpwnam(event.job.euser).pw_uid
+    log_with_caller(pbs.EVENT_DEBUG4, 'UID is %d' % uid)
+    data = {
+        'jobid': jid,
+        'uid': uid
+    }
+    url = HookHelper.build_path(resource='job')
+    timeout = HookHelper.get_config()['post_timeout']
+    try:
+        r = post(url, json=data, timeout=timeout)
+    except requests.Timeout:
+        log_with_caller(pbs.EVENT_ERROR, 'POST timed out for job %s' % jid)
+        raise RejectError()
+    if r.status_code != requests.codes['ok']:
+        log_with_caller(pbs.EVENT_ERROR, 'Invalid status code %d' %
+                        r.status_code)
+        raise RejectError()
+    log_with_caller(pbs.EVENT_DEBUG, 'Job %s registered' % jid)
+
+
+def handle_execjob_end():
+    """
+    Handler for execjob_end events.
+    """
+    log_function_name()
+    jid = HookHelper.get_event().job.id
+    url = HookHelper.build_path(resource='job', jobid=jid)
+    timeout = HookHelper.get_config()['delete_timeout']
+    try:
+        r = delete(url, timeout=timeout)
+        r.raise_for_status()
+    except requests.Timeout:
+        log_with_caller(pbs.EVENT_ERROR, 'POST timed out for job %s' % jid)
+        raise RejectError()
+    except requests.HTTPError:
+        log_with_caller(pbs.EVENT_ERROR, 'DELETE job failed')
+        raise RejectError('Job delete failed')
```

Comment:
You don't handle the timeout like you did above?

Reply:
We talked offline, this is resolved.